src/main/java/com/cloudbees/jenkins/support/impl/HeapUsageHistogram.java

```diff
@@ -0,0 +1,115 @@
+package com.cloudbees.jenkins.support.impl;
+
+import com.cloudbees.jenkins.support.AsyncResultCache;
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.Content;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.model.Node;
+import hudson.remoting.Callable;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+import org.jenkinsci.remoting.RoleChecker;
+import org.kohsuke.accmod.Restricted;
+import org.kohsuke.accmod.restrictions.NoExternalUse;
+
+import javax.management.InstanceNotFoundException;
+import javax.management.MBeanException;
+import javax.management.MBeanServer;
+import javax.management.MalformedObjectNameException;
+import javax.management.ObjectName;
+import javax.management.ReflectionException;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.lang.management.ManagementFactory;
+import java.util.Collections;
+import java.util.Set;
+import java.util.WeakHashMap;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Heap histogram from master node.
+ */
+@Extension
+@Restricted(NoExternalUse.class)
+public class HeapUsageHistogram extends Component {
+    private static final int OFFSET = 3;
+    private static final int MAX = 200 + OFFSET;
```

Comment:
:nit: Variable could be improved. `MAX` what? I assume it's max size, but best to define it as such. Also I'm not sure what's the reason for the `OFFSET` variable other than to set the `MAX` to 203.